cluster-autoscaler/utils/taints/taints.go

```diff
@@ -182,16 +182,25 @@ func MarkDeletionCandidate(node *apiv1.Node, client kube_client.Interface) error
 	return AddTaints(node, client, []apiv1.Taint{taint}, false)
 }
 
+// skipTaintingGridscaleNode checks if a node should not be tainted. This is gridscale specific.
+func skipTaintingGridscaleNode(node *apiv1.Node) (reason string, skip bool) {
+	// The first node should never be tainted
+	if strings.HasSuffix(node.Name, gridscaleNode0SuffixName) {
+		return fmt.Sprintf("Skipping tainting of node %v, because it is a gridscale node 0", node.Name), true
```

Comment:
@fabiante I'd like to make this error more clear:

```suggestion
		return fmt.Sprintf("Skipping tainting of node %v, because the first pool node should never be tained", node.Name), true
```